bot/exts/info/subscribe.py

```diff
@@ -133,15 +148,45 @@ async def update_view(self, interaction: Interaction) -> None:
         self.style = self.REMOVE_STYLE if self.assigned else self.ADD_STYLE
         self.label = self.LABEL_FORMAT.format(action="Remove" if self.assigned else "Add", role_name=self.role.name)
         try:
-            await interaction.message.edit(view=self.view)
+            await interaction.response.edit_message(view=self.view)
         except discord.NotFound:
             log.debug("Subscribe message for %s removed before buttons could be updated", interaction.user)
             self.view.stop()
 
 
+class AllSelfAssignableRolesView(discord.ui.View):
+    """A persistent view that'll hold one button allowing interactors to toggle all available self-assignable roles."""
+
+    def __init__(self, assignable_roles: list[AssignableRole]):
+        super().__init__(timeout=None)
+        self.assignable_roles = assignable_roles
+
+    @discord.ui.button(
+        style=discord.ButtonStyle.success,
+        label="Show all self assignable roles",
+        custom_id="toggle-available-roles-button",
+        row=1
+    )
+    async def show_all_self_assignable_roles(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Sends the original subscription view containing the available self assignable roles."""
+        view = RoleButtonView(interaction.user, self.assignable_roles)
+        await interaction.response.send_message(
+            view=view,
+            ephemeral=True
+        )
+
+
 class Subscribe(commands.Cog):
     """Cog to allow user to self-assign & remove the roles present in ASSIGNABLE_ROLES."""
 
+    GREETING_EMOJI = ":wave:"
+
+    SELF_ASSIGNABLE_ROLES_MESSAGE = (
+        f"Hi there {GREETING_EMOJI},"
+        "\nWe have self-assignable roles for server updates an events!"
```

Comment:
I'm not sure what this sentence is supposed to mean entirely. Perhaps it could be reworded to be concise?

Reply:
That's supposed to be an "and", it's a typo.
Does it still appear unclear to you ?